llvm/lib/Target/AIE/AIE2AddrSpace.h

```diff
@@ -0,0 +1,47 @@
+//===-- AIE2AddrSpace.h - Define Address Space for AIEngine V2 ---*- C++-*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2023-2024 Advanced Micro Devices, Inc. or its affiliates
+//
+//===----------------------------------------------------------------------===//
+//
+// This file declares the AIEngine V2 Address Space and DM banks
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_SUPPORT_AIE2ADDRSPACE_H
+#define LLVM_SUPPORT_AIE2ADDRSPACE_H
+
+namespace llvm {
+
+namespace AIE2AS {
```

Comment:
Nit: I'd prefer
```
namespace AIE2 {

enum class AddressSpaces { ... }

enum class DMBanks { ... }

}
```

This is more in line with what we are doing when writing AIE-version-specific code. Also note I'm using c++ enum classes, not C enums.

Reply:
have updated the enum to c++ enum and they way suggested above.